From 86f998f6ebc800ca3292d699951427b9c54ced75 Mon Sep 17 00:00:00 2001
From: Victor Seva <linuxmaniac@torreviejawireless.org>
Date: Tue, 19 Aug 2014 17:21:57 +0200
Subject: [PATCH] debugger: fix reset_msgid

Don't change the msg->id when that message has been processed already
---
 modules/debugger/debugger_api.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

--- a/modules/debugger/debugger_api.c
+++ b/modules/debugger/debugger_api.c
@@ -270,18 +270,18 @@ int dbg_msgid_filter(struct sip_msg *msg
 	}
 	msgid_base = _dbg_pid_list[indx].msgid_base;
 	lock_release(_dbg_pid_list[indx].lock);
-	msgid_new = msg->id - msgid_base;
-	LM_DBG("msg->id:%d msgid_base:%d -> %d\n", msg->id, msgid_base, msgid_new);
-	if(msgid_new>0)
+	if(msg->id > msgid_base)
 	{
+		msgid_new = msg->id - msgid_base;
+		LM_DBG("msg->id:%d msgid_base:%d -> %d\n",
+			msg->id, msgid_base, msgid_new);
 		msg->id = msgid_new;
-		return 1;
 	}
 	else
 	{
-		LM_WARN("msgid_new<=0??\n");
-		return -1;
+		LM_DBG("msg->id:%d already processed\n", msg->id);
 	}
+	return 1;
 }
 
 /**
